src/plugins/vis_type_table_new/public/utils/use_pagination.ts

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { useCallback, useEffect, useMemo, useState } from 'react';
+import { TableVisConfig } from '../types';
+
+export const usePagination = (visConfig: TableVisConfig, nRow: number) => {
+  const [pagination, setPagination] = useState({
+    pageIndex: 0,
+    pageSize: visConfig.perPage || 0,
+  });
+  const onChangeItemsPerPage = useCallback(
+    (pageSize) => setPagination((p) => ({ ...p, pageSize, pageIndex: 0 })),
+    [setPagination]
+  );
+  const onChangePage = useCallback((pageIndex) => setPagination((p) => ({ ...p, pageIndex })), [
+    setPagination,
+  ]);
+
+  useEffect(() => {
+    const perPage = visConfig.perPage || 0;
+    const maxiPageIndex = Math.ceil(nRow / perPage) - 1;
```

Comment:
it will never be 0. This just want perPage to be number due to ```TableVisParams.perPage: number | ""```

Reply:
Yeah, this isn't right - L23 explicitly allows for it to have a value of `0` if `visConfig.perPage` is falsey (as it will be in the case of the empty string). L24 probably should be type-guarded to ensure it only executes when `perPage` is a positive number.